### DISCUSSION TITLE
DIY Wireless Headset Battery % Widget in KDE Plasma!

### DISCUSSION THREAD

**Discussion by hugeblank** [Sapd/HeadsetControl#201] [Scripts]
I was really happy to find that this tool supports the Logitech G Pro X headset. I've been looking for a way to get some sort of indication of my headsets battery since G HUB isn't available for Linux Operating Systems. Here is a rundown of my little bodge!

**The Process:**
1. Install [HeadsetControl](https://github.com/Sapd/HeadsetControl#compiling). I recommend globally installing, as the little script I provide expects it.
2. Install the [Command Output](https://store.kde.org/p/1166510) Plasma Widget. This can be more easily installed through the plasma UI. Simply right click on desktop, click "Add Widgets", then click "Get New Widgets". In the search bar type "command output". For me it was one of the last results, the one by 'Zren'.
3. Create the script we'll use to write the output. Navigate to the path you'd like to put the script on, and run:
```zsh
wget https://gist.githubusercontent.com/hugeblank/4c494b21a2489077420c2a41f282fff7/raw/5a24f8c8ad1a210dcb78d0a5bb516f8b2ce8df74/battery.sh
chmod 744 battery.sh
```
The gists script is fairly straightforward. Get just the battery percentage, and print it after the headset emoji 🎧:
```zsh
#!/bin/zsh
echo 🎧 `headsetcontrol -b | grep "Battery:" | awk {'printf $2'}`
```
4. Add the widget to your desktop, or to the panel you'd like it to be on. Right click, and click on "Configure Command Output".
5. Change the values to reflect the following:
  - Command: The path to your script
  - Run every: Whatever interval you feel you need the command to update, in milliseconds. I used 5000.
  - Click -> Run Command: The same path to your script. This makes it refresh whenever you left click on the widget, a convenient feature imo.
<img src="https://user-images.githubusercontent.com/16393543/152775417-264dacc4-3746-4571-b13e-5be46620ac91.png" width="600">

**The Result:**
![image](https://user-images.githubusercontent.com/16393543/152771661-6867f4cc-6b1e-4e89-94f2-cb02957b6291.png)
The indicator sitting comfortably next to my other usage widgets! This is the widest the widget generally gets, when you're charging the headset. 

Please feel free to share your own widgets/modifications in the replies to this discussion, I'd love to see what others are doing with HeadsetControl!

I really enjoy the options this program opens up just by being a shell utility. Thank you to the developers and maintainers for working on this project, especially @jaypikay for his recent work on Pro X Wireless Functionality (see #197).

**Comment by hugeblank** [comment]
...and already I've come up with an improvement to the script. While charging, replace the word with a lightning bolt:
```zsh
#!/bin/zsh
batt=`headsetcontrol -b | grep "Battery" | awk {'printf $2'}`
if [[ $batt == "Charging" ]]; then
        batt=🗲
fi
echo 🎧 $batt
```

**Reply by Mr1BitL8r** [reply]
With the newer version 3.0.0-4 of headsetcontrol I needed to use:
```sh
#!/bin/zsh
# Get the real information
battery_status=`headsetcontrol -b | grep "Status" | awk {'printf $2'}`
battery_level=`headsetcontrol -b | grep "Level" | awk {'printf $2'}`

# While charging set the correct value for level (value in % or the charging icon 🗲)
if [[ $battery_status == "BATTERY_CHARGING" ]]; then
  battery_level="🗲"
fi
# Create the output
echo 🎧 $battery_level
```

**Reply by Nama** [reply]
We have json. Also no need to query multiple times. 
```sh
hs_json=$(headsetcontrol -o json)
battery_level=$(echo "$hs_json" | jq -r  '.devices[0].battery.level')
battery_status=$(echo "$hs_json" | jq -r  '.devices[0].battery.status')
```

**Comment by Mr1BitL8r** [comment]
Thank you for the cool idea and solution!
